clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -1393,10 +1376,7 @@ Expected<StringRef> linkDevice(ArrayRef<StringRef> InputFiles,
           "For SPIR targets, Linking is supported only for JIT compilations "
           "and AOT compilations for Intel CPUs/GPUs");
     if (IsSYCLKind) {
-      auto SYCLPostLinkFile = sycl::runSYCLPostLink(InputFiles, Args);
-      if (!SYCLPostLinkFile)
-        return SYCLPostLinkFile.takeError();
-      auto SPVFile = sycl::runLLVMToSPIRVTranslation(*SYCLPostLinkFile, Args);
+      auto SPVFile = sycl::runLLVMToSPIRVTranslation(InputFiles[0], Args);
```

Comment:
sycl-post-link has been moved up as it is required for SYCL offloading for non-spirv targets as well.